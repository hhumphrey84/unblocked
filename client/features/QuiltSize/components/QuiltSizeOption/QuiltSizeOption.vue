<template>
    <li
        :aria-selected="isSelected"
        :class="{ focused: isSelected }"
        :data-test-id="testIds.WRAPPER"
        role="option"
        @click="onClick(data.id)"
    >
        {{ description }} {{ data.width }} x {{ data.height }}
    </li>
</template>

<script>
    import { QUILT_OPTION_TEST_IDS } from '../../quilt-size.constants';

    const props = {
        data: {
            type: Object,
            required: true,
        },
        isSelected: {
            type: Boolean,
            required: true,
        },
        onClick: {
            type: Function,
            required: true,
        },
    };

    const data = {
        testIds: QUILT_OPTION_TEST_IDS,
    };

    const computed = {
        description() {
            return `Some description: ${this.data.id}`;
        },
    };

    export default {
        name: 'QuiltSizeOption',
        props,
        data: () => data,
        computed,
    };
</script>
