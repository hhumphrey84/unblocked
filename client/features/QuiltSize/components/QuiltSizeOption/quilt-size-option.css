@import '../../../../assets/colors.css';
@import '../../../../assets/dimensions.css';
@import '../../../../assets/fonts.css';

.option {
    position: relative;
    padding: calc(var(--spacing) * 2);
    cursor: pointer;

    &.focused,
    &:hover {
        background: var(--color-amazonite);
    }

    &.focused,
    &:hover {
        & .title,
        & .description {
            color: var(--color-white);
        }
    }

    & .title,
    & .description {
        display: block;
        text-transform: uppercase;
    }

    & .title {
        @mixin meta-1;
        padding: 2px 0 4px;
    }

    & .description {
        @mixin meta-2;
        font-size: 12px;
    }
}
