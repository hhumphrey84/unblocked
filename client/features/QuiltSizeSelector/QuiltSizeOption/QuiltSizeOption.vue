<template>
    <li
        :aria-selected="isSelected"
        :class="{ focused: isSelected }"
        role="option"
        @click="onClick(data.id)"
    >
        {{ data.id }} {{ data.width }} x {{ data.height }}
    </li>
</template>

<script>
    const props = {
        data: {
            type: Object,
            required: true,
        },
        isSelected: {
            type: Boolean,
            required: true,
        },
        onClick: {
            type: Function,
            required: true,
        },
    };

    export default {
        name: 'QuiltSizeOption',
        props,
    };
</script>
