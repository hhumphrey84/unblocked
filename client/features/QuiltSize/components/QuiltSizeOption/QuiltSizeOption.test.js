import { shallowMount } from '@vue/test-utils';
import QuiltSizeOption from './QuiltSizeOption.vue';

const optionData = {
    id: '12345',
    width: 10,
    height: 100,
};

const getQuiltSizeOption = ({
    data = optionData,
    isSelected = false,
    onClick = () => null,
} = {}) => shallowMount(QuiltSizeOption, {
    propsData: {
        data,
        isSelected,
        onClick,
    }
});

describe('QuiltSizeOption', () => {
    test('renders properly', () => {
        const wrapper = getQuiltSizeOption();
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('if prop.isSelected is true should add class focused', () => {
        const wrapper = getQuiltSizeOption();

        expect(wrapper.classes('focused')).toBe(false);

        wrapper.setProps({ isSelected: true });

        expect(wrapper.classes('focused')).toBe(true);
    });

    test('if prop.isSelected is true then aria-selected should be true', () => {
        const wrapper = getQuiltSizeOption();

        expect(wrapper.attributes('aria-selected')).toBe(undefined);

        wrapper.setProps({ isSelected: true });

        expect(wrapper.attributes('aria-selected')).toBe('true');
    });

    test('onClick should fire prop.onClick with the data.id', () => {
        const onClickStub = jest.fn();
        const wrapper = getQuiltSizeOption({ onClick: onClickStub });

        wrapper.trigger('click')

        expect(onClickStub.mock.calls.length).toBe(1);
        expect(onClickStub.mock.calls[0][0]).toBe(optionData.id);
    });
})
