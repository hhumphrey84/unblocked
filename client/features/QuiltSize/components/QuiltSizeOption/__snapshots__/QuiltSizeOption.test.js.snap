// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuiltSizeOption renders properly 1`] = `
<li data-test-id="QUILT_SIZE_OPTION" role="option" class="option">
  <h2 class="title">12345</h2>
  <p class="description">10" x 100"</p>
</li>
`;
